### PR TITLE
DAOS-2757 control: add storage prep subcommands to daos_shell

### PR DESCRIPTION
**Pull request by tanabarr**
Add capability for user to prepare NVMe SSDs (setup devices to be used
with SPDK by specified user) and SCM/DCPM modules (create AppDirect
regions and pmem kernel devices) with `daos_shell storage prep-{nvme,scm}`
subcommands.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/cmd/dmg/storage.go [hunk 1]
```diff
@@ -32,9 +32,31 @@ import (
...
+	Format  StorageFormatCmd  `command:"format" alias:"f" description:"Format SCM and NVMe storage attached to remote servers."`
+	Update  StorageUpdateCmd  `command:"fwupdate" alias:"u" description:"Update firmware on NVMe storage attached to remote servers."`
+}
+
+// StoragePrepareCmd is the struct representing the prep storage subcommand.
+type StoragePrepareCmd struct {
+	broadcastCmd
+	connectedCmd
+	common.StoragePrepNvmeCmd
+	common.StoragePrepScmCmd
```
Comment (kjacque) [inline src/control/cmd/dmg/storage.go]: These usages of common should be common_storage, yeah?

File: src/control/server/mgmt_svc.go [hunk 2]
```diff
@@ -238,3 +229,22 @@ func (svc *mgmtSvc) DestroyPool(
 
 	return resp, nil
 }
+
+// KillRank implements the method defined for the Management Service.
+func (svc *mgmtSvc) KillRank(ctx context.Context, req *pb.DaosRank) (*pb.DaosResp, error) {
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/server/mgmt_svc.go]: agreed however this is simply moving placeholder implementation from incorrect location (controlService) to management service, as part of tidy up. functionality is not being added.

File: src/control/server/storage_nvme.go [hunk 3]
```diff
@@ -352,67 +301,53 @@ func (n *nvmeStorage) Format(i int, results *(common.NvmeControllerResults)) {
...
 
 	// appends results to response to provide update specific function
-	addCretUpdate := func(status pb.ResponseStatus, errMsg string) {
-		// log depth should be stack layer registering result
-		*results = append(
-			*results,
-			newCret(
-				"update", pciAddr, status, errMsg,
-				common.UtilLogDepth+1))
+	addCretUpdate := func(status pb.ResponseStatus, errMsg string, infoMsg string) {
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/server/storage_nvme.go]: fixed

File: src/control/server/mgmt_svc.go [hunk 4]
```diff
@@ -17,7 +17,6 @@
 // The Government's rights to use, modify, reproduce, release, perform, display,
 // or disclose this software are subject to the terms of the Apache License as
 // provided in Contract No. 8F-30005.
-// Any reproduction of computer software, computer software documentation, or
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/server/mgmt_svc.go]: fixed

File: src/proto/mgmt/control.proto [hunk 5]
```diff
@@ -27,26 +27,25 @@ package mgmt;
...
-        rpc StorageBurnIn(StorageBurnInReq)
-            returns(stream StorageBurnInResp) {};
-        // Fetch FIO configuration file specifying burn-in jobs/workloads
-        rpc FetchFioConfigPaths(EmptyReq) returns(stream FilePath) {};
-        // Kill a given rank associated with a given pool
-        rpc KillRank(DaosRank) returns(DaosResp) {};
-        // List features supported on remote storage server/DAOS system
-        rpc ListFeatures(EmptyReq) returns(stream Feature) {};
+	// Prepare nonvolatile storage devices for use with DAOS
+	rpc StoragePrepare(StoragePrepareReq) returns (StoragePrepareResp) {};
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/proto/mgmt/control.proto]: fixed

File: src/control/server/mgmt_svc.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kjacque) [reply src/control/server/mgmt_svc.go]: Ah, got it. Didn't notice that it was moved.

**Pull request merged**